common/src/main/java/com/revenuecat/purchases/common/verification/Signature.kt

```diff
@@ -0,0 +1,83 @@
+package com.revenuecat.purchases.common.verification
+
+import android.util.Base64
+import androidx.annotation.VisibleForTesting
+
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+internal enum class SignatureComponent(val size: Int) {
+    INTERMEDIATE_KEY(size = 32),
+    INTERMEDIATE_KEY_EXPIRATION(size = 4),
+    INTERMEDIATE_KEY_SIGNATURE(size = 64),
+    SALT(size = 16),
+    PAYLOAD(size = 64),
+    ;
+
+    companion object {
+        val totalSize: Int
+            get() = values().sumOf { it.size }
+    }
+
+    val startByte: Int
+        get() = values().copyOfRange(0, ordinal).sumOf { it.size }
+
+    val endByte: Int
+        get() = startByte + size
+}
+
+private fun ByteArray.copyOf(component: SignatureComponent): ByteArray {
+    return copyOfRange(component.startByte, component.endByte)
+}
+
+data class Signature(
+    val intermediateKey: ByteArray,
+    val intermediateKeyExpiration: ByteArray,
+    val intermediateKeySignature: ByteArray,
+    val salt: ByteArray,
+    val payload: ByteArray,
+) {
+    companion object {
+        internal fun fromString(signature: String): Signature {
+            val signatureBytes = Base64.decode(signature, Base64.DEFAULT)
+            val expectedSize = SignatureComponent.totalSize
+            if (signatureBytes.size != expectedSize) {
+                throw InvalidSignatureSizeException(
+                    "Invalid signature size. Expected $expectedSize, " +
+                        "got ${signatureBytes.size} bytes",
+                )
+            }
+            return Signature(
+                signatureBytes.copyOf(SignatureComponent.INTERMEDIATE_KEY),
+                signatureBytes.copyOf(SignatureComponent.INTERMEDIATE_KEY_EXPIRATION),
+                signatureBytes.copyOf(SignatureComponent.INTERMEDIATE_KEY_SIGNATURE),
+                signatureBytes.copyOf(SignatureComponent.SALT),
+                signatureBytes.copyOf(SignatureComponent.PAYLOAD),
+            )
+        }
+    }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
When using `ByteArray`, data classes default equals don't seem to work well, so we need to overwrite the `equals` and `hashCode` methods